languages/tree-sitter-stack-graphs-java/src/stack-graphs.tsg

```diff
@@ -0,0 +1,244 @@
+attribute push_symbol = symbol          => type = "push_symbol", symbol = symbol
+attribute scoped_node_definition = node => type = "pop_scoped_symbol", node_symbol = node, is_definition
+attribute scoped_node_reference = node  => type = "push_scoped_symbol", node_symbol = node, is_reference
+attribute symbol_definition = symbol    => type = "pop_symbol", symbol = symbol, is_definition
+attribute symbol_reference = symbol     => type = "push_symbol", symbol = symbol, is_reference
+
+attribute node_symbol = node            => symbol = (source-text node), source_node = node
+
+; ######                                                   
+; #     # #####   ####   ####  #####    ##   #    #  ####  
+; #     # #    # #    # #    # #    #  #  #  ##  ## #      
+; ######  #    # #    # #      #    # #    # # ## #  ####  
+; #       #####  #    # #  ### #####  ###### #    #      # 
+; #       #   #  #    # #    # #   #  #    # #    # #    # 
+; #       #    #  ####   ####  #    # #    # #    #  ####  
+;
+; ########################################################
+
+(program)@prog {
+  node @prog.defs
+  node @prog.lexical_scope
+  edge @prog.lexical_scope -> ROOT_NODE
+  edge ROOT_NODE -> @prog.defs
+}
+
+(program (_)@declaration)@prog {
+  edge @prog.defs -> @declaration.defs
+  edge @declaration.lexical_scope -> @prog.lexical_scope
+}
+
+
+;; Class Expressions
+
+(class_declaration
+  name: (identifier) @name
+  body: (class_body) @class_body) @class {
+
+  node @name.pop
+  node @class.lexical_scope
+  node @class.defs
+  attr (@name.pop) node_definition = @name
+
+  edge @class_body.lexical_scope -> @class.lexical_scope
+
+  edge @class_body.lexical_scope -> @class_body.defs
+  attr (@class_body.lexical_scope -> @class_body.defs) precedence = 1
+  edge @class.defs -> @name.pop
+  edge @name.pop -> @class_body.defs
+}
+
+(class_declaration
+  superclass: (superclass
+    (type_identifier) @identifier) @superclass
+    body: (class_body) @class_body) @class {
+
+    node @identifier.ref
+    attr (@identifier.ref) node_reference = @identifier
+
+    edge @identifier.ref -> @class.lexical_scope
+    edge @class_body.lexical_scope -> @identifier.ref
+}
+
+(class_body) @class_body {
+  node @class_body.lexical_scope
+  node @class_body.defs
+}
+
+(class_body (_)@declaration)@class_body {
+  edge @class_body.defs -> @declaration.defs
+  edge @declaration.lexical_scope -> @class_body.lexical_scope
+}
+
+(method_invocation
+  name: (identifier) @method_name) @method_invocation {
+
+  node @method_name.value
+  attr (@method_name.value) node_reference = @method_name
+  edge @method_name.value -> @method_invocation.before_scope
+  node @method_invocation.defs
+  edge @method_name.value -> @method_invocation.lexical_scope
+}
+
+(method_invocation
+  object: (super) @super
+  name: (identifier) @method_name) @method_invocation {
+}
+
+(method_invocation) @method_invocation
+{
+  node @method_invocation.before_scope
+  node @method_invocation.after_scope
```

Comment:
If `method_invocation` is an expression, the `before_scope` and `after_scope` are unnecessary.

Reply:
I use these nodes elsewhere for some of my tests to pass so I'll need to spend more time on this one to figure out how to remove it and not rely on those edges.